build/azure-pipelines/product-build.yml

```diff
@@ -171,10 +181,108 @@ stages:
             parameters:
               VSCODE_QUALITY: ${{ variables.VSCODE_QUALITY }}
 
+  - ${{ if eq(parameters.VSCODE_BUILD_TUNNEL_CLI, true) }}:
+    - stage: CompileCLI
+      dependsOn: []
+      jobs:
+        - ${{ if eq(variables.VSCODE_BUILD_STAGE_X86_LINUX, true) }}:
+          - job: LinuxX86
+            pool: vscode-1es-linux
+            steps:
+              - template: ./cli/prepare.yml
+                parameters:
+                  VSCODE_QUALITY: ${{ variables.VSCODE_QUALITY }}
+                  VSCODE_IS_POSIX: true
+              - template: ./cli/compile-linux.yml
+                parameters:
+                  VSCODE_CLI_DIR: $(Build.SourcesDirectory)/cli
+                  VSCODE_CLI_BINARY_NAME: ${{ variables.VSCODE_CLI_BINARY_NAME }}
+                  VSCODE_CLI_TARGETS:
+                    - ${{ if eq(parameters.VSCODE_BUILD_LINUX_ALPINE, true) }}:
+                      - { target: x86_64-unknown-linux-musl, artifact: vscode_cli_alpine_x64_cli-unsigned }
```

Comment:
I've adjusted these slightly, but I still think passing in artifact names is helpful here.

The alternative I see is passing individual flags like `VSCODE_CLI_BUILD_LINUX_X64_MUSL` and doing a big set of if/else's inside the pipeline to pick the target and artifact, but I feel like doing it here is more succinct -- especially since later `product-build-*.yml`'s need to know the artifact names to pull them down as well.